// ng2-loading-spinner/esm2015/lib/config.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Inject, Injectable, Optional } from '@angular/core';
import { ANIMATION_TYPES } from './animation-types';
import { findIndex, objectValues } from './utils';
export class ConfigService {
    /**
     * @param {?} config
     */
    constructor(config) {
        this.config = config;
        this.config = this.config || {};
        this.defaultConfig = {
            animationType: this.config.animationType || ANIMATION_TYPES.fadingCircle,
            backdropColor: this.config.backdropColor || 'rgba(0, 0, 0, 0.3)',
            spinnerColor: this.config.spinnerColor || '#fff',
            spinnerPosition: this.config.spinnerPosition || 'center',
            backdropBorderRadius: this.config.backdropBorderRadius || '0',
            spinnerSize: this.config.spinnerSize || 'md',
            spinnerFontSize: this.config.spinnerFontSize || ''
        };
    }
    /**
     * @param {?} config
     * @return {?}
     */
    normalizeConfigs(config) {
        if (!config) {
            config = this.defaultConfig;
            return config;
        }
        if (config.spinnerSize === '' && config.spinnerFontSize === '') {
            config.spinnerFontSize = '1rem';
        }
        for (const option in this.defaultConfig) {
            if (!config[option]) {
                config[option] = this.defaultConfig[option];
            }
        }
        if (findIndex(objectValues(ANIMATION_TYPES), config['animationType']) === -1) {
            config['animationType'] = ANIMATION_TYPES.fadingCircle;
        }
        return config;
    }
}
ConfigService.decorators = [
    { type: Injectable },
];
/** @nocollapse */
ConfigService.ctorParameters = () => [
    { type: undefined, decorators: [{ type: Inject, args: ['loadingConfig',] }, { type: Optional }] }
];
if (false) {
    /**
     * @type {?}
     * @private
     */
    ConfigService.prototype.defaultConfig;
    /**
     * @type {?}
     * @private
     */
    ConfigService.prototype.config;
}
