// ng2-loading-spinner/esm5/lib/config.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/**
 * @record
 */
export function INg2LoadingSpinnerConfig() { }
if (false) {
    /** @type {?|undefined} */
    INg2LoadingSpinnerConfig.prototype.animationType;
    /** @type {?|undefined} */
    INg2LoadingSpinnerConfig.prototype.backdropColor;
    /** @type {?|undefined} */
    INg2LoadingSpinnerConfig.prototype.backdropBorderRadius;
    /** @type {?|undefined} */
    INg2LoadingSpinnerConfig.prototype.spinnerColor;
    /** @type {?|undefined} */
    INg2LoadingSpinnerConfig.prototype.spinnerPosition;
    /** @type {?|undefined} */
    INg2LoadingSpinnerConfig.prototype.spinnerSize;
    /** @type {?|undefined} */
    INg2LoadingSpinnerConfig.prototype.spinnerFontSize;
}
var Ng2LoadingSpinnerConfig = /** @class */ (function () {
    function Ng2LoadingSpinnerConfig(config) {
        this.config = config;
        this.animationType = config.animationType;
        this.backdropColor = config.backdropColor;
        this.spinnerColor = config.spinnerColor;
        this.spinnerPosition = config.spinnerPosition;
        this.backdropBorderRadius = config.backdropBorderRadius;
        this.spinnerSize = config.spinnerSize;
        this.spinnerFontSize = config.spinnerFontSize;
    }
    return Ng2LoadingSpinnerConfig;
}());
export { Ng2LoadingSpinnerConfig };
if (false) {
    /** @type {?} */
    Ng2LoadingSpinnerConfig.prototype.animationType;
    /** @type {?} */
    Ng2LoadingSpinnerConfig.prototype.backdropColor;
    /** @type {?} */
    Ng2LoadingSpinnerConfig.prototype.spinnerColor;
    /** @type {?} */
    Ng2LoadingSpinnerConfig.prototype.spinnerPosition;
    /** @type {?} */
    Ng2LoadingSpinnerConfig.prototype.backdropBorderRadius;
    /** @type {?} */
    Ng2LoadingSpinnerConfig.prototype.spinnerSize;
    /** @type {?} */
    Ng2LoadingSpinnerConfig.prototype.spinnerFontSize;
    /**
     * @type {?}
     * @private
     */
    Ng2LoadingSpinnerConfig.prototype.config;
}
