// ng2-loading-spinner/esm5/lib/ng2-loading-spinner.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { Ng2LoadingSpinnerComponent } from './ng2-loading-spinner.component';
import { Ng2LoadingSpinnerDirective } from './ng2-loading-spinner.directive';
var Ng2LoadingSpinnerModule = /** @class */ (function () {
    function Ng2LoadingSpinnerModule() {
    }
    /**
     * @param {?} globalConfig
     * @return {?}
     */
    Ng2LoadingSpinnerModule.forRoot = /**
     * @param {?} globalConfig
     * @return {?}
     */
    function (globalConfig) {
        return {
            ngModule: Ng2LoadingSpinnerModule,
            providers: [{ provide: 'loadingConfig', useValue: globalConfig }]
        };
    };
    Ng2LoadingSpinnerModule.decorators = [
        { type: NgModule, args: [{
                    imports: [
                        CommonModule
                    ],
                    declarations: [Ng2LoadingSpinnerComponent, Ng2LoadingSpinnerDirective],
                    exports: [Ng2LoadingSpinnerDirective],
                    entryComponents: [Ng2LoadingSpinnerComponent]
                },] },
    ];
    return Ng2LoadingSpinnerModule;
}());
export { Ng2LoadingSpinnerModule };
