// ng2-loading-spinner/esm5/lib/ng2-loading-spinner.directive.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { ComponentFactoryResolver, Directive, ElementRef, Input, Renderer2, TemplateRef, ViewContainerRef } from '@angular/core';
import { Ng2LoadingSpinnerComponent } from './ng2-loading-spinner.component';
import { ConfigService } from './config.service';
var Ng2LoadingSpinnerDirective = /** @class */ (function () {
    function Ng2LoadingSpinnerDirective(el, vcRef, cfResolver, renderer, configService) {
        this.el = el;
        this.vcRef = vcRef;
        this.cfResolver = cfResolver;
        this.renderer = renderer;
        this.configService = configService;
    }
    /**
     * @return {?}
     */
    Ng2LoadingSpinnerDirective.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        this.setPosition();
    };
    /**
     * @param {?} changes
     * @return {?}
     */
    Ng2LoadingSpinnerDirective.prototype.ngOnChanges = /**
     * @param {?} changes
     * @return {?}
     */
    function (changes) {
        if (changes.show) {
            if (changes.show.currentValue) {
                this.createSpinner();
            }
            else {
                this.destroySpinner();
            }
        }
    };
    /**
     * @return {?}
     */
    Ng2LoadingSpinnerDirective.prototype.ngOnDestroy = /**
     * @return {?}
     */
    function () {
        this.destroySpinner();
    };
    /**
     * @return {?}
     */
    Ng2LoadingSpinnerDirective.prototype.setPosition = /**
     * @return {?}
     */
    function () {
        /** @type {?} */
        var elPosition = this.el.nativeElement.style.position;
        if (elPosition === 'relative' || elPosition === 'absolute') {
            return;
        }
        this.el.nativeElement.style.position = 'relative';
    };
    /**
     * @return {?}
     */
    Ng2LoadingSpinnerDirective.prototype.createSpinner = /**
     * @return {?}
     */
    function () {
        /** @type {?} */
        var spinnerCF = this.cfResolver.resolveComponentFactory(Ng2LoadingSpinnerComponent);
        this.spinnerComponentRef = this.vcRef.createComponent(spinnerCF);
        this.config = this.configService.normalizeConfigs(this.config);
        this.spinnerComponentRef.instance.config = this.config;
        this.spinnerComponentRef.instance.template = this.template;
        this.renderer.appendChild(this.vcRef.element.nativeElement, this.spinnerComponentRef.injector.get(Ng2LoadingSpinnerComponent).vcRef.element.nativeElement);
    };
    /**
     * @return {?}
     */
    Ng2LoadingSpinnerDirective.prototype.destroySpinner = /**
     * @return {?}
     */
    function () {
        if (this.spinnerComponentRef) {
            this.spinnerComponentRef.destroy();
        }
    };
    Ng2LoadingSpinnerDirective.decorators = [
        { type: Directive, args: [{
                    selector: '[ng2-loading]',
                    providers: [ConfigService]
                },] },
    ];
    /** @nocollapse */
    Ng2LoadingSpinnerDirective.ctorParameters = function () { return [
        { type: ElementRef },
        { type: ViewContainerRef },
        { type: ComponentFactoryResolver },
        { type: Renderer2 },
        { type: ConfigService }
    ]; };
    Ng2LoadingSpinnerDirective.propDecorators = {
        show: [{ type: Input, args: ['ng2-loading',] }],
        config: [{ type: Input }],
        template: [{ type: Input }]
    };
    return Ng2LoadingSpinnerDirective;
}());
export { Ng2LoadingSpinnerDirective };
if (false) {
    /** @type {?} */
    Ng2LoadingSpinnerDirective.prototype.show;
    /** @type {?} */
    Ng2LoadingSpinnerDirective.prototype.config;
    /** @type {?} */
    Ng2LoadingSpinnerDirective.prototype.template;
    /**
     * @type {?}
     * @private
     */
    Ng2LoadingSpinnerDirective.prototype.spinnerComponentRef;
    /**
     * @type {?}
     * @private
     */
    Ng2LoadingSpinnerDirective.prototype.el;
    /**
     * @type {?}
     * @private
     */
    Ng2LoadingSpinnerDirective.prototype.vcRef;
    /**
     * @type {?}
     * @private
     */
    Ng2LoadingSpinnerDirective.prototype.cfResolver;
    /**
     * @type {?}
     * @private
     */
    Ng2LoadingSpinnerDirective.prototype.renderer;
    /**
     * @type {?}
     * @private
     */
    Ng2LoadingSpinnerDirective.prototype.configService;
}
